### ISSUE TITLE
Can't download ITV hub show

### ISSUE THREAD

**Issue by seanadl** [ytdl-org/youtube-dl#30265]
## Checklist

- [x] I'm asking a question
- [x] I've looked through the README and FAQ for similar questions
- [x] I've searched the bugtracker for similar questions including closed ones

## Question

I am trying to download a show from the ITV hub (here: https://www.itv.com/hub/the-ellen-degeneres-show/2a4350a1024). Using Stream Detector on Firefox, I get the following code to paste into Youtube-dl:

`youtube-dl --no-part --restrict-filenames --user-agent "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0" --add-header "Cookie:Itv.Session={%22tokens%22:{%22content%22:{%22entitlement%22:{%22purchased%22:[]%2C%22failed_availability_checks%22:[]%2C%22source%22:%22%22}%2C%22email_verified%22:true%2C%22access_token%22:%22eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJodHRwczovL2F1dGguaXR2LmNvbSIsInN1YiI6IjZkYmFmYWU3LTU0NzYtNDNmYS1hYWEwLTAwODQxYmY2OTFkMSIsImV4cCI6MTYzNzg0MjMwMiwiaWF0IjoxNjM3NzUyMzAyLCJicm9hZGNhc3RFcnJvck1zZyI6IiIsImJyb2FkY2FzdFJlc3BvbnNlQ29kZSI6IjIwMCIsImJyb2FkY2FzdGVyIjoiSVRWIiwiaXNBY3RpdmUiOnRydWUsIm5vbmNlIjoiZGdSQ1gzWWVkNURvWWl6VjhQN20iLCJuYW1lIjoiU2VhbiIsInNjb3BlIjoiY29udGVudCIsImVudGl0bGVtZW50cyI6W10sInBheW1lbnRTb3VyY2UiOiIiLCJzaG93UHJpdmFjeU5vdGljZSI6ZmFsc2UsInVuZGVyMTgiOmZhbHNlfQ.Da_C_X8gHGWh6QBo1Im_hqspwvo9z8XVKkItpCqKW9BYtJVivSSB5eqO-b1Q9U8fk9QN5IqXTM2nOHe22BUTm02rcKUXgsTJ41pRqaAuAquuSXkPpZSnOgbiE0flsEnQN78mfTzs0k9jvLO-A0QT939NowHCcd8_BfbdFCrE11FI0zCFyTHKatVuS9j1bDK98Nd2O61iphtBcmNBwaql1p94okb6QWYSI3kqFyNBICgziTwi-6t-kCZ76_wSopIPUuQ21oRz4911esS3kBId4SdrUcSpezAgUS-_GS3XpCMiOLWGJ69fXU3qzkXP8NRLYSxcf_RTU4QkmWs6kjvaXg%22%2C%22token_type%22:%22bearer%22%2C%22refresh_token%22:%22eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJodHRwczovL2F1dGguaXR2LmNvbSIsInN1YiI6IjZkYmFmYWU3LTU0NzYtNDNmYS1hYWEwLTAwODQxYmY2OTFkMSIsImV4cCI6NDc5MzQyNTkwMiwiaWF0IjoxNjM3NzUyMzAyLCJub25jZSI6ImRnUkNYM1llZDVEb1lpelY4UDdtIiwic2NvcGUiOiJjb250ZW50IiwiYXV0aF90aW1lIjoxNjM3NzUyMzAyfQ.GV1cqako7yaGvA393O9oa9cVhX0I-Sw5M2wSR1gg5RAYt1lzYIY0T2nLUu3xLte8p1JqlAVb_1IEV_Yts12dxQTHVmDupqA33dKIlA0Pxxkq1Mv8x5WjjOuVgnmjf1lmm7Ygt8560V9olis5fKlONa4QNrsX1c4O0RNr630WzPBbFk6e1PsGJ2bVOBxp5ja8kzun-_oOCjonrQOxTHj3-e9RmyCVJdbRLzpFkfJf_n03zNXly_d-uBad3vodfHemv8dQJeXXKSDQGVAq3idlPj77bEHfRkfmE6yUMBjMlZSF_C3HfqomFyN84wNw9JiQLMumwP2tHNpfnYVvqBWy4Q%22}}%2C%22sticky%22:true}; mid=1709081696670734686; _cs_c=1; _cs_id=8fcd982a-e516-ad59-e9f0-5ec3cfcfacdc.1637751312.1.1637752301.1637751312.1.1671915312583; _cs_s=7.0.0.1637754101993; __utma=187186800.658762206.1637751313.1637751313.1637751313.1; __utmb=187186800.13.9.1637752166735; __utmc=187186800; __utmz=187186800.1637751313.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none); _hjSessionUser_1586194=eyJpZCI6IjcyMmJhMjBmLWU5MzEtNTA1YS1iYjYyLWFhMTI3NTBkMTViOSIsImNyZWF0ZWQiOjE2Mzc3NTEzMTMwODcsImV4aXN0aW5nIjp0cnVlfQ==; _hjFirstSeen=1; _hjSession_1586194=eyJpZCI6IjQ2MTY2NDM1LThhZTQtNGJhOC05NjAzLTM4ZTBjOGVjMjY1MiIsImNyZWF0ZWQiOjE2Mzc3NTEzMTMwOTN9; _hjAbsoluteSessionInProgress=0; Itv.CookiePolicy.v2=accepted; Itv.Region=ITV|null; __utmt_UA-17825253-53=1" --referer "https://www.itv.com/" "https://itvpnpdotcom.cdn1.content.itv.com/2-4350-1024-001/18/1/VAR028/2-4350-1024-001_18_1_VAR028.ism/.mpd?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9pdHZwbnBkb3Rjb20uY2RuMS5jb250ZW50Lml0di5jb20vMi00MzUwLTEwMjQtMDAxLzE4LzEvVkFSMDI4LzItNDM1MC0xMDI0LTAwMV8xOF8xX1ZBUjAyOC5pc20vKiIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTYzNzc3NTcwM319fV19&Signature=InPnv~NiUQzV2St09QxkG5u63iiUWKbWsV4V5csiShiz0YOEVf9gy7OvYAPRCPXjSg2T95wzshNY8Whl92I2AMBv69jyyrY0s8zGI7ja6dMxZB0mbxyg5L8M15Y2TnwwgTSnfqpDfX9HkZdBZXnhRt~Ko6cbPf-Ao8NcI0-DcNFKrAuw8AQIECo-F4YFdECk9AcqPcgWvIVJngUYhQmQY97foz2~13UhTK7H-DbGqyYyocgt-YDeNfBCnLfoBotVDqe2q6laK81wX5BGHiEKzj44t3hHqE-H4GUhfq8vHeHYbyeR92pDSjAlNGuW4CYxQqXRTQl70fI6dN~PP32rKg__&Key-Pair-Id=APKAJB7PCFZAZHWZVIBA"`

However, this produces the following error: 

`ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.`

**Comment by dirkf** [comment]
Patch your yt-dl from PR #28955 and use it directly. Or use some other tool designed to download the media URL, maybe _ffmpeg_.

However IIRC some bought-in ITV shows, possibly including Ellen, are DRM-protected and so the show may not be accessible even if you can download the linked media.

**Comment by seanadl** [comment]
Ah thanks. I'll at least try to download it and see if it's DRM-protected. 

How do I patch yt-dl on a mac?

**Comment by dirkf** [comment]
Depending on how you installed it, find the `youtube_dl/extractor/itv.py` file and replace it with the version from the PR. If the directory is not directly writable by your user account, also run `youtube-dl --version` with admin/su permissions to overwrite any previous byte-code compiled version.

**Comment by seanadl** [comment]
Thanks. I hate to keep asking, but not sure how to do the above. I've searched through my system files for that file, but can't find anything. I suspect I need to be doing something entirely different in order to find and replace that file.

**Comment by 89z** [comment]
@seanadl If you can provide credentials, I can work on this. Just run the following command and comment back with the output:

~~~
echo USER:PASS | age -a -r age1zkz9g0kld97vs826j0r7p3ghvz3zcyd4tcqr2vrxpt0uux7mkycsl77hy6
~~~

https://github.com/FiloSottile/age

**Comment by dirkf** [comment]
> Thanks. I hate to keep asking, but not sure how to do the above.

How did you install yt-dl?

```youtube-dl -v -F ' https://www.itv.com/hub/the-ellen-degeneres-show/2a4350a1024'```

```ls -al `which youtube-dl` ```

Etc.

**Comment by Vangelis66** [comment]
@dirkf wrote: 

> Patch your yt-dl from PR #28955 and use it directly.

... Am afraid that PR has been **invalidated since yesterday**, Nov 23rd 2021, because the ITVHub "people" started talking a la "master Yoda" (from Star Wars) :smile: ... 

See issue https://github.com/yt-dlp/yt-dlp/issues/1775 and today's fix, https://github.com/yt-dlp/yt-dlp/pull/1776, implemented in 

https://github.com/yt-dlp/yt-dlp/commit/da27aee

I, as much as others I suppose 😉 , would greatly value a _similar fix_ on top of PR #28955... @sleaux-meaux (original author of that PR) or, perhaps, @dirkf (also UK-based, with a good - and highly appreciated 👍 - record of providing patches to many `yt-dl` breakages 😉 ) ? 

> However IIRC some bought-in ITV shows, possibly **including Ellen**, are DRM-protected

This is my recollection, too; "The Ellen DeGeneres Show" on ITV's VOD service has **always** been **DRM-only** 😞 ... 
In the first post, @seanadl is trying to download (via `yt-dl`) an MPEG-DASH manifest (.mpd) : 
```
https://itvpnpdotcom.cdn1.content.itv.com/2-4350-1024-001/18/1/VAR028/2-4350-1024-001_18_1_VAR028.ism/.mpd?authToken
```
(token has expired by now), which, in all probability, is DRM protected (**cenc**) ... 
All familiar with itvIE already know that if no HLSe streams (targeting older mobile devices) are available for a certain show, then, sadly, it's a **no go** for downloading apps (breaking DRM won't be discussed here...).

**Comment by seanadl** [comment]
> > Thanks. I hate to keep asking, but not sure how to do the above.
> 
> How did you install yt-dl?
> 
> `youtube-dl -v -F 'https://www.itv.com/hub/the-ellen-degeneres-show/2a4350a1024'`
> 
> `` ls -al `which youtube-dl` ``
> 
> Etc.

When I put the above in, i get this: -rwxr-xr-x  1 root  admin  1840311 24 Nov 11:02 /usr/local/bin/youtube-dl

**Comment by dirkf** [comment]
You have a Python script archive version, a !# line followed by a zip archive.

```
 sed < /usr/local/bin/youtube-dl >youtube-dl.zip -n -e '/^#!/ n; p;'
```
Replace the `itv.py` file (once we've fixed it!) in `youtube-dl.zip`; then
```
 { sed < /usr/local/bin/youtube-dl -n -e '/^#!/ p; q;'; cat youtube-dl.zip; }>youtube-dl.new
```
Finally install `youtube-dl.new` as `/usr/local/bin/youtube-dl` (with execute permission).

**Comment by seanadl** [comment]
I'm getting this error when running the above command:

`sed: RE error: illegal byte sequence`

**Comment by dirkf** [comment]
Try the updated version.

**Comment by seanadl** [comment]
> Try the updated version.

That worked. Just waiting for the updated itv.py file and then i can test it out

**Comment by dirkf** [comment]
Here is https://github.com/ytdl-org/youtube-dl/pull/30266.

As far as I can see, the API request for test #1 (Plebs S1E1) gives 404 unless you're logged in, but I'd be happy to add a different test or fix the extractor to make the test work.

However, test #2 (J Ross S17E08) runs, except that the test skips the download (M3U8, expected).

Bad news for OP. As suggested, the extractor identifies the Ellen episode as 'No downloads available', because the available media types are:
```
{
  'mobile': [
  ],
  'dotcom': [
    ['mpeg-dash',
    'outband-webvtt',
    'playready'],
    [
      'mpeg-dash',
      'widevine'
    ],
    [
      'fairplay',
      'hls',
      'outband-webvtt'
    ],
    [
      'mpeg-dash',
      'outband-webvtt',
      'widevine'
    ],
    [
      'fairplay',
      'hls'
    ],
    [
      'mpeg-dash',
      'playready'
    ]
  ]
}
```
So no non-DRM formats.

**Comment by Vangelis66** [comment]
> Here is PR #30266, which basically back-ports the `yt-dlp` extractor for **ITV**.

Congrats; works as expected! 👍 
```
youtube-dl --no-geo-bypass -F "https://www.itv.com/hub/an-audience-with/L0055a0007" -v => 

[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--no-geo-bypass', '-F', 'https://www.itv.com/hub/an-audience-with/L0055a0007', '-v']
[debug] Encodings: locale cp1253, fs utf-8, out utf-8, pref cp1253
[debug] youtube-dl version 2021.06.06+18-git-20210701-ga803582+PRs#30184,#30266
[debug] Lazy loading extractors enabled
[debug] Python version 3.7.12 (CPython) - Windows-Vista-6.0.6003-SP2
[debug] exe versions: ffmpeg 4.4.1, ffprobe 4.4.1, phantomjs 2.1.1, rtmpdump 2.4

[debug] Proxy map: {}
[ITV] L0055a0007: Downloading webpage
[ITV] L0055a0007: Downloading JSON metadata
[ITV] L0055a0007: Downloading m3u8 information
[info] Available formats for L0055a0007:
format code  extension  resolution note
hls-68       mp4        audio only   68k , mp4a.40.2@ 64k
hls-102      mp4        audio only  102k , mp4a.40.2@ 96k
hls-285      mp4        512x288     285k , avc1.4D401F@ 203k, 25.0fps, mp4a.40.2@ 64k
hls-636      mp4        512x288     636k , avc1.4D401F@ 503k, 25.0fps, mp4a.40.2@ 96k
hls-848      mp4        512x288     848k , avc1.4D401F@ 703k, 25.0fps, mp4a.40.2@ 96k
hls-1272     mp4        896x504    1272k , avc1.4D401F@1103k, 25.0fps, mp4a.40.2@ 96k
hls-1908     mp4        896x504    1908k , avc1.4D401F@1703k, 25.0fps, mp4a.40.2@ 96k (best)
```

**Comment by seanadl** [comment]
> Finally install `youtube-dl.new` as `/usr/local/bin/youtube-dl` (with execute permission).

How do I do that part of the install? Done everything else so far!

**Comment by dirkf** [comment]
Probably [s/t like](https://ss64.com/osx/install.html):
```
sudo install -p -o root youtube-dl.new /usr/local/bin/youtube-dl
```

**Comment by dirkf** [comment]
> As far as I can see, the API request for test `#1` (Plebs S1E1) gives 404 unless you're logged in, but I'd be happy to add a different test or fix the extractor to make the test work.

This is weird.
* `youtube-dl  -v -F 'https://www.itv.com/hub/plebs/2a1873a0002'` (Python 3.5) generally works;
* `python3.5 -m youtube_dl  -v -F 'https://www.itv.com/hub/plebs/2a1873a0002'` (in the dev directory) sometimes works;
* `python3.5 test/test_download.py TestDownload.test_ITV` (same URL, in the dev directory) sometimes works
* `python2.7 -m youtube_dl  -v -F 'https://www.itv.com/hub/plebs/2a1873a0002'` (in the dev directory) never works
* `python2.7 test/test_download.py TestDownload.test_ITV` (same URL, in the dev directory) never works.

**Comment by seanadl** [comment]
Hmm. Since updating, I now get this error when trying to use youtube-dl:

`/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python: can't find '__main__' module in '/usr/local/bin/youtube-dl'
`

**Comment by dirkf** [comment]
The self-extracting archive has gone awry. Try the process again.

Also check this output for `/usr/local/bin/youtube-dl`:
```
$ dd if=youtube-dl.new count=1 bs=64 status=none | od -a 
0000000   #   !   /   u   s   r   /   b   i   n   /   e   n   v  sp   p
0000020   y   t   h   o   n  nl   P   K etx eot dc4 nul nul nul  bs nul
0000040  sp  bs   !   (   p   }   h   }   3 dc1 nul nul stx   ? nul nul
0000060 dc1 nul  fs nul   y   o   u   t   u   b   e   _   d   l   /   a
0000100
$
```

**Comment by seanadl** [comment]
Maybe I did something wrong when trying to update the  itv.py file. I unzipped, opened the file and pasted in the contents of the updated itv.py file, re-zipped it back up. Then ran:
` { head -n 1 < /usr/local/bin/youtube-dl; cat youtube-dl.zip; }>youtube-dl.new
`

**Comment by seanadl** [comment]
> | age -a -r age1zkz9g0kld97vs826j0r7p3ghvz3zcyd4tcqr2vrxpt0uux7mkycsl77hy6

Here is what I get:
`YWdlLWVuY3J5cHRpb24ub3JnL3YxCi0+IFgyNTUxOSBBcEJVTDNoWS9JbnhWa0t5
MnA4YWFNL3FrZkhpL2FyaDZSaHROaFBvSW5FCmxZeDlJUXgwbGtvNHZOdHFTU3Va
VWMyZjF4K2JPbllCbHVqTnJiZ3B3RlUKLS0tIEQyWi9nWm4vdExnakJ6ZGloZXgy
MHpRa0h0Z2UxdXQwaS9CVmpYRnZmSEEKGsbNOd51mCWzJbUcbMTLeEo7UaljKt84
29QazEQhUU6RE3+r/jgSa4gr4cbGWtqL4aNrjTGD37DpBz9MfJOgEPs=`

This is the actual video I'm trying to download: https://www.itv.com/hub/the-ellen-degeneres-show/2a4350a1027

**Comment by dirkf** [comment]
We've [already established that you won't be able to download a playable version of this show](https://github.com/ytdl-org/youtube-dl/issues/30265#issuecomment-978154243) using yt-dl.

For patching the installation, it looks like the re-compression didn't work properly. IIRC Finder's Compress doesn't have an update-in-place function, unlike perhaps something like iZip. The actual script used to create the Unix self-extracting version of yt-dl is in the [Makefile](https://github.com/ytdl-org/youtube-dl/blob/master/Makefile); it's meant to be run in the `youtube-dl` folder of a checked out version of the repository:
```
youtube-dl: youtube_dl/*.py youtube_dl/*/*.py
	mkdir -p zip
	for d in youtube_dl youtube_dl/downloader youtube_dl/extractor youtube_dl/postprocessor ; do \
	  mkdir -p zip/$$d ;\
	  cp -pPR $$d/*.py zip/$$d/ ;\
	done
	touch -t 200001010101 zip/youtube_dl/*.py zip/youtube_dl/*/*.py
	mv zip/youtube_dl/__main__.py zip/
	cd zip ; zip -q ../youtube-dl youtube_dl/*.py youtube_dl/*/*.py __main__.py
	rm -rf zip
	echo '#!$(PYTHON)' > youtube-dl
	cat youtube-dl.zip >> youtube-dl
	rm youtube-dl.zip
	chmod a+x youtube-dl
`
where `PYTHON=/usr/bin/env python` by default.

**Comment by 89z** [comment]
@seanadl please paste the output directly as is, do not try to be clever and modify the output.

**Comment by seanadl** [comment]
> @seanadl please paste the output directly as is, do not try to be clever and modify the output.

Hmm. I might not have completely understood the instruction. I just tried doing exactly what you said. I replaced USER: PASS with the ITV player user/pass (as I assume that's what i needed to do) using the following command: 

`echo USER:PASS | age -a -r age1zkz9g0kld97vs826j0r7p3ghvz3zcyd4tcqr2vrxpt0uux7mkycsl77hy6`

And the output is this: 

`-----BEGIN AGE ENCRYPTED FILE-----
YWdlLWVuY3J5cHRpb24ub3JnL3YxCi0+IFgyNTUxOSBBcEJVTDNoWS9JbnhWa0t5
MnA4YWFNL3FrZkhpL2FyaDZSaHROaFBvSW5FCmxZeDlJUXgwbGtvNHZOdHFTU3Va
VWMyZjF4K2JPbllCbHVqTnJiZ3B3RlUKLS0tIEQyWi9nWm4vdExnakJ6ZGloZXgy
MHpRa0h0Z2UxdXQwaS9CVmpYRnZmSEEKGsbNOd51mCWzJbUcbMTLeEo7UaljKt84
29QazEQhUU6RE3+r/jgSa4gr4cbGWtqL4aNrjTGD37DpBz9MfJOgEPs=
-----END AGE ENCRYPTED FILE-----`

I'm sorry im not tech savvy. Was replacing USER:PASS with my actual ITV player user/pass the right or wrong thing to do?

**Comment by 89z** [comment]
> Hmm. I might not have completely understood the instruction. I just tried doing exactly what you said.

OK the corrected output works. However it seems in addition to the login, ITV is region restricted as well:

> We've detected that you're outside of the UK. Live TV and Catch Up are only
available for viewing in the UK, as of December 31st 2020.If you are an ITV
Hub+ subscriber you can watch already downloaded content. Please visit our Help
Site for more information.

Not to mention, even if I was in the UK, I would still likely have to deal with the problem mentioned by @dirkf. So in short, thats not going to be a video that I can get (or that anyone can get easily).

**Comment by Ulrar** [comment]
> > Here is PR #30266, which basically back-ports the `yt-dlp` extractor for **ITV**.
> 
> Congrats; works as expected! 👍

Same here, that PR works perfectly.
Unfortunate that the ITV Hub seems to only have very low resolution videos, but at least it works. Thanks !

**Comment by 89z** [comment]
Not sure why people are saying the PR works, as it doesnt work, even with YT-DLP directly:

~~~
PS C:\> yt-dlp https://www.itv.com/hub/plebs/2a1873a0002
[ITV] 2a1873a0002: Downloading webpage
[ITV] 2a1873a0002: Downloading JSON metadata
ERROR: [ITV] 2a1873a0002: Unable to download JSON metadata: HTTP Error 403:
Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on
https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version;
see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp
with the --verbose flag and include its complete output. (caused by <HTTPError
403: 'Forbidden'>); please report this issue on
https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version;
see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp
with the --verbose flag and include its complete output.

PS C:\> yt-dlp --version
2022.01.21
~~~

**Comment by Vangelis66** [comment]
@89z : 

If I'm not mistaken, you're located in the **US**, but **ITVHub** is a service limited to **UK** IPs only (actually, to a fraction of the whole UK, e.g. Scottish IPs may be blacklisted; in Scotland, **STV** has the rights of most ITV content...).

For many years, geo-blockage could have been circumvented by an `X-Forwarded-For` request header "hack" (which is **still there** inside the **itvIE**), in order for their **playlist API** to spit out streams even from overseas IPs, additionally the stream CDN(s) would not geo-block... Those were the happy times, so to speak...

However, ITV have strengthened their geo-fencing schemes considerably, first by imposing geo-restriction on the **actual stream CDN** (so that an overseas user wasn't able to download the stream even if the API was "fooled" to generate a master HLS playlist), and secondly, more recently, they even fully geo-restricted **access to the playlist API itself** (i.e. the X-F-F trick doesn't work anymore), which is what you are getting requesting **JSON metadata** from the US: 
```
ERROR: [ITV] 2a1873a0002: Unable to download JSON metadata: HTTP Error 403: 
Forbidden (caused by <HTTPError 403: 'Forbidden'>);
```
However, if inside **England** or via using a **whitelisted UK VPN**, the "story" is different: 
```
yt-dlp -F "https://www.itv.com/hub/plebs/2a1873a0002" => 

[ITV] 2a1873a0002: Downloading webpage
[ITV] 2a1873a0002: Downloading JSON metadata
[ITV] 2a1873a0002: Downloading m3u8 information
[info] Available formats for 2a1873a0002:
ID       EXT RESOLUTION FPS |   FILESIZE   TBR PROTO  | VCODEC        VBR ACODEC     ABR
----------------------------------------------------------------------------------------
hls-136  mp4 audio only     | ~ 22.48MiB  136k m3u8_n | audio only        mp4a.40.2 128k
hls-326  mp4 512x288     50 | ~ 53.89MiB  326k m3u8_n | avc1.4D401E  179k mp4a.40.2 128k
hls-613  mp4 512x288     50 | ~101.32MiB  613k m3u8_n | avc1.4D401F  449k mp4a.40.2 128k
hls-813  mp4 512x288     50 | ~134.38MiB  813k m3u8_n | avc1.4D401F  638k mp4a.40.2 128k
hls-1105 mp4 896x504     50 | ~182.65MiB 1105k m3u8_n | avc1.4D401F  914k mp4a.40.2 128k
hls-1694 mp4 896x504     50 | ~280.01MiB 1694k m3u8_n | avc1.4D401F 1469k mp4a.40.2 128k
```
So please, don't jump to conclusions ("**as it doesnt work**") simply based on your own findings...

**Comment by 89z** [comment]
> So please, don't jump to conclusions ("as it doesnt work") simply based on your own findings...

@Vangelis66 its a valid conclusion. I ran everything as one would be expected to do, and I got unexpected result. If I had run my command, and the output was "ITV is only available for viewing in the UK" (like you get with the website), then I would say that the YouTube-DL/YT-DLP is working correctly. But for me to run the exact command, and get a 403 with no explanation to the root cause, is a mistake in the code. If an item suffers from regional lockout, the user should be notified of that, same as what the website does.
